aiohttp_pydantic/injectors.py

```diff
@@ -70,7 +71,14 @@ async def inject(self, request: BaseRequest, args_view: list, kwargs_view: dict)
                 text='{"error": "Malformed JSON"}', content_type="application/json"
             ) from None
 
-        kwargs_view[self.arg_name] = self.model(**body)
+        # Pydantic tries to cast certain structures, such as a list of 2-tuples,
+        # to a dict. Prevent this by requiring the body to be a dict for object models.
+        if self._expect_object and not isinstance(body, dict):
+            raise HTTPBadRequest(
+                text='{"error": "Malformed JSON"}', content_type="application/json"
```

Comment:
The json is not Malformed it's just a list instead of dict please use this text instead of Malformed error 

raise HTTPBadRequest(
        text='[{"in": "body", "loc": ["__root__"], "msg": "value is not a valid dict", "type": "type_error.dict"}]',
        content_type="application/json")